model/atmosphere/dycore/tests/dycore/integration_tests/test_solve_nonhydro.py

```diff
@@ -1745,6 +1746,305 @@ def test_apply_divergence_damping_and_update_vn(
+        z_vn_avg.asnumpy()[:, :],
+        rtol=1.0e-6,
+    )
+
+    # same tolerances as in Liskov
+    assert helpers.dallclose(
+        z_graddiv_vn_ref.asnumpy()[:, :],
+        z_graddiv_vn.asnumpy()[:, :],
+        rtol=1.0e-2,
+        atol=1.0e-20,
+    )
+
+    assert helpers.dallclose(
+        vt_ref.asnumpy()[:, :],
+        vt.asnumpy()[:, :],
+        rtol=1.0e-6,
+    )
+
+    assert helpers.dallclose(
+        mass_fl_e_ref.asnumpy()[:, :],
+        mass_fl_e.asnumpy()[:, :],
+        rtol=1.0e-6,
+    )
+
+    assert helpers.dallclose(
+        z_theta_v_fl_e_ref.asnumpy()[:, :],
+        z_theta_v_fl_e.asnumpy()[:, :],
+        rtol=1.0e-6,
+    )
+
+    assert helpers.dallclose(
+        vn_ie_ref.asnumpy()[:, :],
+        vn_ie.asnumpy()[:, :],
+        rtol=1.0e-5,
+    )
+
+    assert helpers.dallclose(
+        z_vt_ie_ref.asnumpy()[:, :],
+        z_vt_ie.asnumpy()[:, :],
+        rtol=1.0e-6,
+    )
+
+    assert helpers.dallclose(
+        z_kin_hor_e_ref.asnumpy()[:, :],
+        z_kin_hor_e.asnumpy()[:, :],
+        rtol=1.0e-6,
+    )
+
+    assert helpers.dallclose(
+        z_w_concorr_me_ref.asnumpy()[:, :],
+        z_w_concorr_me.asnumpy()[:, :],
+        rtol=1.0e-7,
+    )
+
+
+@pytest.mark.embedded_remap_error
+@pytest.mark.datatest
+@pytest.mark.parametrize("at_first_substep, istep_init, istep_exit", [(True, 2, 2)])
+@pytest.mark.parametrize(
+    "experiment, step_date_init, step_date_exit",
+    [
+        (
+            dt_utils.REGIONAL_EXPERIMENT,
+            "2021-06-20T12:00:10.000",
+            "2021-06-20T12:00:10.000",
+        ),
+        (
+            dt_utils.GLOBAL_EXPERIMENT,
+            "2000-01-01T00:00:02.000",
+            "2000-01-01T00:00:02.000",
+        ),
+    ],
+)
+def test_compute_averaged_vn_and_fluxes_and_prepare_tracer_advection(
+    istep_init,
+    istep_exit,
+    substep_init,
+    substep_exit,
+    step_date_init,
+    step_date_exit,
+    experiment,
+    icon_grid,
+    ndyn_substeps,
+    at_first_substep,
+    grid_savepoint,
+    savepoint_dycore_30_to_38_init,
+    savepoint_dycore_30_to_38_exit,
+    interpolation_savepoint,
+    metrics_savepoint,
+    savepoint_nonhydro_init,
+    savepoint_nonhydro_exit,
+    backend,
+):
+    edge_domain = h_grid.domain(dims.EdgeDim)
+
+    ddqz_z_full_e = metrics_savepoint.ddqz_z_full_e()
+    config = utils.construct_solve_nh_config(experiment)
+
+    z_vn_avg = savepoint_dycore_30_to_38_init.z_vn_avg()
+    mass_fl_e = savepoint_dycore_30_to_38_init.mass_fl_e()
+    z_theta_v_fl_e = savepoint_dycore_30_to_38_init.z_theta_v_fl_e()
+    vn_traj = savepoint_nonhydro_init.vn_traj()
+    mass_flx_me = savepoint_nonhydro_init.mass_flx_me()
+    e_flx_avg = interpolation_savepoint.e_flx_avg()
+    vn = savepoint_dycore_30_to_38_init.vn()
+    z_rho_e = savepoint_dycore_30_to_38_init.z_rho_e()
+    z_theta_v_e = savepoint_dycore_30_to_38_init.z_theta_v_e()
+    r_nsubsteps = 1.0 / ndyn_substeps
```

Comment:
is this nowhere in the config or solve_nonhydro? I feel like it should be defined there, not imported from the outside